demos/reference-check/components/FormWithElements.tsx

```diff
@@ -1,39 +1,50 @@
 import React, { FormEvent, useState } from 'react';
 import { TextElement, useBasisTheory } from '@basis-theory/basis-theory-react';
-import { Box, Button, TextField, useTheme } from '@mui/material';
+import { LoadingButton } from '@mui/lab';
+import { Box, TextField, useTheme } from '@mui/material';
 import { INTER_FONT, PHONE_NUMBER_MASK } from '@/components/constants';
 import { ttl } from '@/components/utils';
 
 export const FormWithElements = () => {
   const [name, setName] = useState('');
+  const [isPhoneNumberComplete, setPhoneNumberComplete] = useState(false);
+  const [loading, setLoading] = useState(false);
   const { bt } = useBasisTheory();
   const theme = useTheme();
 
   const submit = async (event: FormEvent) => {
     event.preventDefault();
-    const phoneNumber = bt?.getElement('phoneNumber');
-    const token = await bt?.tokens.create({
-      // id: '{{ data | alias_preserve_format }}',
-      type: 'token',
-      data: phoneNumber,
-      expiresAt: ttl(),
-    });
+    setLoading(true);
 
-    await fetch('/api/drivers', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({
-        name,
-        phoneNumber: token.id,
-        tokenized: true,
-      }),
-    });
-    setName('');
-    phoneNumber?.clear();
+    try {
+      const phoneNumber = bt?.getElement('phoneNumber');
+      const token = await bt?.tokens.create({
+        // id: '{{ data | alias_preserve_format }}',
+        type: 'token',
+        data: phoneNumber,
+        expiresAt: ttl(),
+      });
+
+      await fetch('/api/drivers', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          name,
+          phoneNumber: token.id,
```

Comment:
if `bt` isn't initialized in time, it looks like this will throw an error. we can either catch or add a condition to see if bt is initialized

Reply:
I don't know how big of a deal this is since this _should_ only get executed when a user submits the form

Reply:
That is correct. We disable the submit button based on the phone being complete, which only happens if the element is filled. I will add a safeguard to the function anyways. Thanks